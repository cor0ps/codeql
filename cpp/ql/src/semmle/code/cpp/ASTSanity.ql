/**
 * @name AST Sanity Check
 * @description Performs sanity checks on the Abstract Syntax Tree. This query should have no results.
 * @kind table
 * @id cpp/ast-sanity-check
 */

import cpp
import CastSanity
