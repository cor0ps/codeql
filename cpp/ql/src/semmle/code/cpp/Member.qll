import semmle.code.cpp.Element
import semmle.code.cpp.Type
