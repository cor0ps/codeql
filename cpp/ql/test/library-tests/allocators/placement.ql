import cpp

from NewOrNewArrayExpr new
select new, new.getPlacementPointer() as placement
