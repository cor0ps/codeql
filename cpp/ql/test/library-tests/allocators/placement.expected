| allocators.cpp:129:3:129:21 | new | allocators.cpp:129:7:129:13 | & ... |
| allocators.cpp:132:3:132:17 | new[] | allocators.cpp:132:7:132:9 | buf |
